File: app/infrastructure/schemas/informacion_empresa_entity.py
from typing import Optional
from pydantic import BaseModel


class InformacionEmpresaEntityRag(BaseModel):
    empresa: Optional[str] = None
    perfil: Optional[str] = None
    seniority: Optional[str] = None
    pais: Optional[str] = None
    informacion_empresa_vect: Optional[list[str]] = None

File: src/main/java/com/entrevistador/analizadorempresa/infrastructure/adapter/entity/InformacionEmpresaEntity.java
package com.entrevistador.analizadorempresa.infrastructure.adapter.entity;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.List;

@Getter
@Setter
@Builder
@Document(collection = "informacion_empresa")
@NoArgsConstructor
@AllArgsConstructor
public class InformacionEmpresaEntity {
    @Id
    private String uuid;
    private String empresa;
    private String perfil;
    private String seniority;
    private String pais;
    private String descripcionVacante;
    private List<EntrevistaEntity> informacionEmpresaVect;
}
